### ISSUE TITLE
Tdarr detecting files renamed by Plugin as new files

### ISSUE THREAD

**Issue by CplPwnies** [HaveAGitGat/Tdarr#390]
**Describe the bug**
When a plugin renames a file, Tdarr detects that file as a new file.

**To Reproduce**
Use a plugin that renames files

**Expected behavior**
The file to be renamed without being removed, readded, and subsequently re-scanned (healthcheck and transcode - even though transcode does nothing because it already meets all requirements)

**Screenshots**
N/A

**Please provide the following information:**
**Plugin Code**
```
/* eslint-disable */
function details() {
    return {
        id: "Tdarr_Local_Plugin_Rename",
        Stage: "Post-processing",
        Name: "Rename based on resolution and append to end of file for rescans",
        Type: "Video",
        Operation: "",
        Description: "Rename files based on resolution and append tag to end of filename for *arr rescans",
        Version: "1.00",
        Link: "",
        Tags: "post-processing"
    };
}

function plugin(file, librarySettings, inputs) {
    var response = {
        file,
        removeFromDB: false,
        updateDB: false,
        reQueueAfter: false,
        infoLog: ""
    };
    
    try {
        var fs = require('fs');
        var path = require('path');
        
        if(fs.existsSync(path.join(process.cwd(), '/node_modules'))) {
            var rootModules = path.join(process.cwd(), '/node_modules/');
        } else {
            var rootModules = '';
        }
        
        var fsextra = require(rootModules + "fs-extra");
        var fileNameOld = file._id;
        
        var tvRegex = /S[0-9]+E[0-9]+/g;
        var tvMatches = file._id.match(tvRegex);
        
        if (file.video_resolution === '720p' && file._id.includes('1080p') && tvMatches.length >= 1) {
            file._id = file._id.replace('1080p', '720p');
            file.file = file.file.replace('1080p', '720p');
        }
        
        var fileNameArr = file._id.split('.');
        if (fileNameArr[fileNameArr.length-3] != "h265") {
            if (fileNameArr[fileNameArr.length-3] == "MLP" || fileNameArr[fileNameArr.length-3] == "DTS-HD" || fileNameArr[fileNameArr.length-2] == "Atmos") {
                response.infoLog += '[-] MLP/DTS/Atmos Audio Codec detected! Using -4 instead.\n'
                if (fileNameArr[fileNameArr.length-4] != "h265") {
                    fileNameArr[fileNameArr.length-4] = "h265";
                    file._id = fileNameArr.join('.');
                    file.file = fileNameArr.join('.');
                }
            } else {
                fileNameArr[fileNameArr.length-3] = "h265";
                file._id = fileNameArr.join('.');
                file.file = fileNameArr.join('.');
            }
        } else { }

        if (fileNameOld != file._id) {
            response.infoLog += `[-] Old Name: ${fileNameOld}\n`;
            response.infoLog += `[-] New Name: ${file._id}\n`;
            fsextra.moveSync(fileNameOld, file._id, { overwrite: true });
            response.updateDB = true;
            response.file = file;
        } else {
            response.infoLog += '[+] No Rename required!\n';
        }
        
    }
    catch(err) { response.infoLog += `${err}\n`; console.log(`[DEBUG] UH-OH: ${err}`); }
    return response;
}

module.exports.dependencies = [
  'fs-extra',
];
module.exports.details = details;
module.exports.plugin = plugin;
```

**Logs**
This is shown in the logs after file rename:

> [2021-03-01T12:56:44.292] [INFO] Tdarr_Server - 4nRcXRNq-: file removed, removing:/media/tvshows/MyShow/Season 06/MyShow.S06E06.WEBDL-480p.h264.EAC3.mkv
> [2021-03-01T12:56:44.295] [INFO] Tdarr_Server - 4nRcXRNq-: file removed, removing:/media/tvshows/MyShow/Season 06/MyShow.S06E08.WEBDL-480p.h264.EAC3.mkv
> [2021-03-01T12:56:44.296] [INFO] Tdarr_Server - 4nRcXRNq-: file removed, removing:/media/tvshows/MyShow/Season 06/MyShow.S06E09.WEBDL-480p.h264.EAC3.mkv
> [2021-03-01T12:56:54.312] [INFO] Tdarr_Server - 4nRcXRNq-: File detected, adding to queue:/media/tvshows/MyShow/Season 06/MyShow.S06E08.WEBDL-480p.h265.EAC3.mkv
> [2021-03-01T12:56:54.312] [INFO] Tdarr_Server - 4nRcXRNq-: File detected, adding to queue:/media/tvshows/MyShow/Season 06/MyShow.S06E06.WEBDL-480p.h265.EAC3.mkv
> [2021-03-01T12:56:54.312] [INFO] Tdarr_Server - 4nRcXRNq-: File detected, adding to queue:/media/tvshows/MyShow/Season 06/MyShow.S06E09.WEBDL-480p.h265.EAC3.mkv

 - OS: Debian 10 Buster
 - Browser: Chrome
 - Version: Docker - Latest

**Additional context**
See my plugin code above and log details. This happens every time the plug renames a file and ultimately, all the plugin does is rename the codec to reflect h265, and change the resolution if it is a TV show and 1080p.

You can see that the files are the same in this example except for the codec.

**Comment by HaveAGitGat** [comment]
This hasn't really been an issue in the past as like you say the new file doesn't need to have any transcoding done on it but will try and add a fix for 08.

**Comment by CplPwnies** [comment]
That's great! The part that bothers me the most about this is that I don't get to see the stats on the file, it just looks like this after all is said and done:
![image](https://user-images.githubusercontent.com/4989931/110952926-63ac1400-830c-11eb-8d3c-bef0656e35c5.png)

**Comment by HaveAGitGat** [comment]
I've fixed this for 2.00.08
